<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
  <head>
    <p><a id="logo" href="http://scala-lang.org/"><img src="/sites/default/files/icons/Scala_Logo2008.png" alt="scala-lang.org" border=0/></a></p>
    <title id="Title">Aladdin - Scala Bugtracking</title>
        <style type="text/css">
        table {
            font-size: 10pt;
            background-color: #ebb85c;
        }
        tr.impair        { background-color: #f0be64 }
        tr.impairContrib { background-color: #f0be64 }

#buraq-content {
	float: none;
	color: black;
	background-color: white;
	font-size: 1.25em;
	margin-left: 9.6em;
}
        </style>
      </head>
  <body><!--DisplayBugsBlock_Start-->




<table WIDTH="100%" cellspacing="4" cellpadding="2">

	<tr><td></td>
		<th>[#1288]</th>
		<th><strong>project:</strong> compiler</th>
		<th><strong>priority:</strong> medium</th>
		<th><strong>category:</strong> bug</th>
        </tr>

        <tr>
          <td></td>
	  <th align="center">submitter</th>
	  <th align="center">assigned to</th>
	  <th align="center">status</th>
	  <th align="center">date submitted</th>
        </tr>

        <tr>
          <td></td>
	  <td align="center" bgcolor="#FFFFFF">Stephane</td>
	  <td align="center" bgcolor="#FFFFFF">Burak</td>
	  <td align="center" bgcolor="#FFFFFF">moved to trac</td>
          <td align="center" bgcolor="#FFFFFF">2007-08-23 13:13:04.0</td>

	</tr>

        <tr>
		<th align="right">subject</th>
		<td colspan="4" bgcolor="#FFFFFF">[contrib #756] Exception catching code</td>
	</tr>
	<tr>	
		<th align="right">code</th>
		<td colspan="4" bgcolor="#FFFFFF">
	<font size="-1"><pre>The new exception matching code works great (thank you!) but I recently
found another case that fails.  This is with 2.6.0-RC1.

The following code shows that all exceptions match the first catch
clause.  This ticket may be considered a follow up to

http://scala-webapps.epfl.ch/bugtracking/contribs/display.do?id=649
http://scala-webapps.epfl.ch/bugtracking/bugs/displayItem.do?id=1168

Regards,
Blair

class MyException1 extends Exception

// Commenting out the following line and uncommenting the second line
// will cause the test to succeed.
trait SpecialException extends MyException1
// trait SpecialException

class MyException2 extends MyException1 with SpecialException

object Test extends Application
{
  Array[Throwable](new Exception(&quot;abc&quot;),
                   new MyException1,
                   new MyException2).foreach { e =&gt;
    try {
      throw e
    }
    catch {
      case e : SpecialException =&gt; {
        assume(e.isInstanceOf[SpecialException])
      }
      case e =&gt; {
        assume(e.isInstanceOf[Throwable])
      }
    }
  }
}
</pre></font></td>
	</tr>
	<tr>
		<th style="text-align:right;">what happened</th>
		<td colspan="4" style="background-color:#FFFFFF;" ><pre>java.lang.Error: assumption failed
        at scala.Predef$.assume(Predef.scala:101)
        at Test$$anonfun$0.apply(t2.scala:21)
        at Test$$anonfun$0.apply(t2.scala:15)
        at scala.Iterator$class.foreach(Iterator.scala:375)
        at scala.runtime.BoxedArray$$anon$0.foreach(BoxedArray.scala:46)
        at scala.Iterable$class.foreach(Iterable.scala:241)
        at scala.runtime.BoxedArray.foreach(BoxedArray.scala:25)
        at Test$.<init>(t2.scala:15)
        at Test$.<clinit>(t2.scala)
        at Test.main(t2.scala)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
        at java.lang.reflect.Method.invoke(Method.java:597)
        at scala.tools.nsc.ObjectRunner$$anonfun$0.apply(ObjectRunner.scala:75)
        at scala.tools.nsc.ObjectRunner$.withContextClassLoader(ObjectRunner.scala:49)
        at scala.tools.nsc.ObjectRunner$.run(ObjectRunner.scala:74)
        at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:154)
        at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
</pre></td>
	</tr>
	<tr>
		<th align="right">what expected</th>
		<td colspan="4" bgcolor="#FFFFFF" >No assertions</td>
	</tr>
	<tr>
	
  <td colspan="4" align="right">
      
      <a href="displayBugs.do.html">
        [back to overview]</a></td></tr>

</table>

  <tr>
          <td align="center" colspan="2">
            <strong>Changes of this bug report</strong>
          </td>
        </tr><table><tr>
        <td>
        <strong>Stephane&nbsp; edited on &nbsp;2007-08-23 13:13:37.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>

</td></tr>
<tr>
        <td>
        <strong>Gilles&nbsp; edited on &nbsp;2007-08-29 09:54:44.0</strong>
        </td>
        </tr>
<tr><td bgcolor='#FFFFFF'>

</td></tr></table>
<!--DisplayBugsBlock_End-->  </body>
</html>